diff --git a/app/Resources/views/profile/account.html.twig b/app/Resources/views/profile/account.html.twig
index c9dfe0e..3bcdff9 100644
--- a/app/Resources/views/profile/account.html.twig
+++ b/app/Resources/views/profile/account.html.twig
@@ -32,7 +32,11 @@
                         <tr>
                             <th>Credit Card</th>
                             <td>
-                                None
+                                {% if app.user.cardBrand %}
+                                    {{ app.user.cardBrand }} ending in {{ app.user.cardLast4 }}
+                                {% else %}
+                                    None
+                                {% endif %}
                             </td>
                         </tr>
                     </tbody>
diff --git a/src/AppBundle/Controller/OrderController.php b/src/AppBundle/Controller/OrderController.php
index d36dcec..53abb79 100644
--- a/src/AppBundle/Controller/OrderController.php
+++ b/src/AppBundle/Controller/OrderController.php
@@ -87,11 +87,19 @@ class OrderController extends BaseController
         /** @var User $user */
         $user = $this->getUser();
         if (!$user->getStripeCustomerId()) {
-            $stripeClient->createCustomer($user, $token);
+            $stripeCustomer = $stripeClient->createCustomer($user, $token);
         } else {
-            $stripeClient->updateCustomerCard($user, $token);
+            $stripeCustomer = $stripeClient->updateCustomerCard($user, $token);
         }
 
+        // save card details!
+        $cardDetails = $stripeCustomer->sources->data[0];
+        $user->setCardBrand($cardDetails->brand);
+        $user->setCardLast4($cardDetails->last4);
+        $em = $this->getDoctrine()->getManager();
+        $em->persist($user);
+        $em->flush();
+
         $cart = $this->get('shopping_cart');
 
         foreach ($cart->getProducts() as $product) {
diff --git a/src/AppBundle/StripeClient.php b/src/AppBundle/StripeClient.php
index 4784158..411b375 100644
--- a/src/AppBundle/StripeClient.php
+++ b/src/AppBundle/StripeClient.php
@@ -37,6 +37,8 @@ class StripeClient
 
         $customer->source = $paymentToken;
         $customer->save();
+
+        return $customer;
     }
 
     public function createInvoiceItem($amount, User $user, $description)
