diff --git a/src/AppBundle/Controller/ProfileController.php b/src/AppBundle/Controller/ProfileController.php
index 1b2e799..bbe754c 100644
--- a/src/AppBundle/Controller/ProfileController.php
+++ b/src/AppBundle/Controller/ProfileController.php
@@ -28,6 +28,12 @@ class ProfileController extends BaseController
         $stripeClient = $this->get('stripe_client');
         $stripeClient->cancelSubscription($this->getUser());
 
+        $subscription = $this->getUser()->getSubscription();
+        $subscription->deactivateSubscription();
+        $em = $this->getDoctrine()->getManager();
+        $em->persist($subscription);
+        $em->flush();
+
         $this->addFlash('success', 'Subscription Canceled :(');
 
         return $this->redirectToRoute('profile_account');
diff --git a/src/AppBundle/Entity/Subscription.php b/src/AppBundle/Entity/Subscription.php
index 6265a00..5bdc706 100644
--- a/src/AppBundle/Entity/Subscription.php
+++ b/src/AppBundle/Entity/Subscription.php
@@ -99,4 +99,11 @@ class Subscription
         $this->billingPeriodEndsAt = $periodEnd;
         $this->endsAt = null;
     }
+
+    public function deactivateSubscription()
+    {
+        // paid through end of period
+        $this->endsAt = $this->billingPeriodEndsAt;
+        $this->billingPeriodEndsAt = null;
+    }
 }
