diff --git a/app/Resources/views/profile/account.html.twig b/app/Resources/views/profile/account.html.twig
index 0c3cee5..cca7695 100644
--- a/app/Resources/views/profile/account.html.twig
+++ b/app/Resources/views/profile/account.html.twig
@@ -10,7 +10,9 @@
 
                     {% if app.user.hasActiveSubscription %}
                         {% if app.user.subscription.isCancelled %}
-                            todo - add reactivate button
+                            <form action="{{ path('account_subscription_reactivate') }}" method="POST" class="pull-right">
+                                <button type="submit" class="btn btn-success btn-xs">Reactivate Subscription</button>
+                            </form>
                         {% else %}
                             <form action="{{ path('account_subscription_cancel') }}" method="POST" class="pull-right">
                                 <button type="submit" class="btn btn-danger btn-xs">Cancel Subscription</button>
@@ -29,10 +31,6 @@
                                         <span class="label label-warning">Cancelled</span>
 
                                         But you have access until {{ app.user.subscription.endsAt|date('F jS') }}
-
-                                        <form action="{{ path('account_subscription_reactivate') }}" method="POST" style="display: inline;">
-                                            <button type="submit" class="btn btn-success btn-xs">Reactive Subscription</button>
-                                        </form>
                                     {% else %}
                                         <span class="label label-success">Active</span>
                                     {% endif %}
diff --git a/src/AppBundle/Controller/ProfileController.php b/src/AppBundle/Controller/ProfileController.php
index 55e9dd0..d05423a 100644
--- a/src/AppBundle/Controller/ProfileController.php
+++ b/src/AppBundle/Controller/ProfileController.php
@@ -45,6 +45,13 @@ class ProfileController extends BaseController
     public function reactivateSubscriptionAction()
     {
         $stripeClient = $this->get('stripe_client');
-        $stripeClient->reactivateSubscription($this->getUser());
+        $stripeSubscription = $stripeClient->reactivateSubscription($this->getUser());
+
+        $this->get('subscription_helper')
+            ->addSubscriptionToUser($stripeSubscription, $this->getUser());
+
+        $this->addFlash('success', 'Welcome back!');
+
+        return $this->redirectToRoute('profile_account');
     }
 }
