diff --git a/app/Resources/views/profile/account.html.twig b/app/Resources/views/profile/account.html.twig
index f9a82ac..f309e9e 100644
--- a/app/Resources/views/profile/account.html.twig
+++ b/app/Resources/views/profile/account.html.twig
@@ -10,7 +10,9 @@
 
                     {% if app.user.hasActiveSubscription %}
                         {% if app.user.subscription.isCancelled %}
-                            todo - add reactivate button
+                            <form action="{{ path('account_subscription_reactivate') }}" method="POST" class="pull-right">
+                                <button type="submit" class="btn btn-success btn-xs">Reactivate Subscription</button>
+                            </form>
                         {% else %}
                             <form action="{{ path('account_subscription_cancel') }}" method="POST" class="pull-right">
                                 <button type="submit" class="btn btn-danger btn-xs">Cancel Subscription</button>
diff --git a/src/AppBundle/Controller/ProfileController.php b/src/AppBundle/Controller/ProfileController.php
index bbe754c..55e9dd0 100644
--- a/src/AppBundle/Controller/ProfileController.php
+++ b/src/AppBundle/Controller/ProfileController.php
@@ -38,4 +38,13 @@ class ProfileController extends BaseController
 
         return $this->redirectToRoute('profile_account');
     }
+
+    /**
+     * @Route("/profile/subscription/reactivate", name="account_subscription_reactivate")
+     */
+    public function reactivateSubscriptionAction()
+    {
+        $stripeClient = $this->get('stripe_client');
+        $stripeClient->reactivateSubscription($this->getUser());
+    }
 }
diff --git a/src/AppBundle/StripeClient.php b/src/AppBundle/StripeClient.php
index 4b06968..30f3bd0 100644
--- a/src/AppBundle/StripeClient.php
+++ b/src/AppBundle/StripeClient.php
@@ -85,4 +85,20 @@ class StripeClient
             'at_period_end' => true,
         ]);
     }
+
+    public function reactivateSubscription(User $user)
+    {
+        if (!$user->hasActiveSubscription()) {
+            throw new \LogicException('Subscriptions can only be reactivated if the subscription has not actually ended yet');
+        }
+
+        $subscription = \Stripe\Subscription::retrieve(
+            $user->getSubscription()->getStripeSubscriptionId()
+        );
+        // this triggers the refresh of the subscription!
+        $subscription->plan = $user->getSubscription()->getStripePlanId();
+        $subscription->save();
+
+        return $subscription;
+    }
 }
