diff --git a/src/AppBundle/Controller/OrderController.php b/src/AppBundle/Controller/OrderController.php
index 752340c..4452db0 100644
--- a/src/AppBundle/Controller/OrderController.php
+++ b/src/AppBundle/Controller/OrderController.php
@@ -35,7 +35,18 @@ class OrderController extends BaseController
         if ($request->isMethod('POST')) {
             $token = $request->request->get('stripeToken');
 
-            dump($token);
+            \Stripe\Stripe::setApiKey("XXX_PRIVATEKEY_XXX");
+            \Stripe\Charge::create(array(
+              "amount" => $this->get('shopping_cart')->getTotal() * 100,
+              "currency" => "usd",
+              "source" => $token,
+              "description" => "First test charge!"
+            ));
+
+            $this->get('shopping_cart')->emptyCart();
+            $this->addFlash('success', 'Order Complete! Yay!');
+
+            return $this->redirectToRoute('homepage');
         }
 
         return $this->render('order/checkout.html.twig', array(
