diff --git a/app/Resources/views/order/checkout.html.twig b/app/Resources/views/order/checkout.html.twig
index 5eae347..24b30ab 100644
--- a/app/Resources/views/order/checkout.html.twig
+++ b/app/Resources/views/order/checkout.html.twig
@@ -37,10 +37,11 @@
                     <script
                         src="https://checkout.stripe.com/checkout.js" class="stripe-button"
                         data-key="pk_test_HxZzNHy8LImKK9LDtgMDRBwd"
-                        data-amount="999"
+                        data-amount="{{ cart.total * 100 }}"
                         data-name="Dollar Shear Club"
                         data-description="Widget"
-                        data-image="/img/documentation/checkout/marketplace.png"
+                        data-image="{{ asset('images/logo.png') }}"
+                        data-zip-code="true"
                         data-locale="auto">
                     </script>
                 </form>
