diff --git a/src/AppBundle/Controller/OrderController.php b/src/AppBundle/Controller/OrderController.php
index 746995b..752340c 100644
--- a/src/AppBundle/Controller/OrderController.php
+++ b/src/AppBundle/Controller/OrderController.php
@@ -6,6 +6,7 @@ use AppBundle\Entity\Product;
 use Sensio\Bundle\FrameworkExtraBundle\Configuration\Method;
 use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
 use Sensio\Bundle\FrameworkExtraBundle\Configuration\Security;
+use Symfony\Component\HttpFoundation\Request;
 
 class OrderController extends BaseController
 {
@@ -27,10 +28,16 @@ class OrderController extends BaseController
      * @Route("/checkout", name="order_checkout")
      * @Security("is_granted('ROLE_USER')")
      */
-    public function checkoutAction()
+    public function checkoutAction(Request $request)
     {
         $products = $this->get('shopping_cart')->getProducts();
 
+        if ($request->isMethod('POST')) {
+            $token = $request->request->get('stripeToken');
+
+            dump($token);
+        }
+
         return $this->render('order/checkout.html.twig', array(
             'products' => $products,
             'cart' => $this->get('shopping_cart')
