diff --git a/README.md b/README.md
index 09cc070..d937bff 100644
--- a/README.md
+++ b/README.md
@@ -15,8 +15,13 @@ php composer.phar install
 ```
 
 This will interactively ask you for some settings - like `database_host`.
-Fill those in. If you make a mistake - just modify `app/config/parameters.yml`
-afterwards!
+Fill those in.
+
+This will *also* ask you about some **Stripe** keys. If you haven't already,
+signup for a new [Stripe](https://stripe.com/) account. You can find the
+secret and public/publishable keys in your "Account Settings" area.
+
+If you make a mistake - just modify `app/config/parameters.yml` afterwards!
 
 If you get an error about "Unknown Database" - it's cool! Ignore it for now :)
 
diff --git a/app/Resources/views/order/checkout.html.twig b/app/Resources/views/order/checkout.html.twig
index 24b30ab..585e4ef 100644
--- a/app/Resources/views/order/checkout.html.twig
+++ b/app/Resources/views/order/checkout.html.twig
@@ -36,7 +36,7 @@
                 <form action="" method="POST">
                     <script
                         src="https://checkout.stripe.com/checkout.js" class="stripe-button"
-                        data-key="pk_test_HxZzNHy8LImKK9LDtgMDRBwd"
+                        data-key="{{ stripe_public_key }}"
                         data-amount="{{ cart.total * 100 }}"
                         data-name="Dollar Shear Club"
                         data-description="Widget"
diff --git a/app/config/parameters.yml.dist b/app/config/parameters.yml.dist
index 0087ffc..69e8170 100644
--- a/app/config/parameters.yml.dist
+++ b/app/config/parameters.yml.dist
@@ -17,3 +17,5 @@ parameters:
 
     # A secret key that's used to generate certain security-related tokens
     secret:            ThisTokenIsNotSoSecretChangeIt
+    stripe_secret_key: XXX_PRIVATEKEY_XXX
+    stripe_public_key: YYY_PUBLISHABLE_KEY_YYY
diff --git a/src/AppBundle/Controller/OrderController.php b/src/AppBundle/Controller/OrderController.php
index 4452db0..88af599 100644
--- a/src/AppBundle/Controller/OrderController.php
+++ b/src/AppBundle/Controller/OrderController.php
@@ -35,7 +35,7 @@ class OrderController extends BaseController
         if ($request->isMethod('POST')) {
             $token = $request->request->get('stripeToken');
 
-            \Stripe\Stripe::setApiKey("XXX_PRIVATEKEY_XXX");
+            \Stripe\Stripe::setApiKey($this->getParameter('stripe_secret_key'));
             \Stripe\Charge::create(array(
               "amount" => $this->get('shopping_cart')->getTotal() * 100,
               "currency" => "usd",
@@ -51,7 +51,8 @@ class OrderController extends BaseController
 
         return $this->render('order/checkout.html.twig', array(
             'products' => $products,
-            'cart' => $this->get('shopping_cart')
+            'cart' => $this->get('shopping_cart'),
+            'stripe_public_key' => $this->getParameter('stripe_public_key')
         ));
 
     }
