diff --git a/src/AppBundle/Controller/OrderController.php b/src/AppBundle/Controller/OrderController.php
index a700780..03f55bb 100644
--- a/src/AppBundle/Controller/OrderController.php
+++ b/src/AppBundle/Controller/OrderController.php
@@ -91,8 +91,14 @@ class OrderController extends BaseController
             return $this->redirectToRoute('order_checkout');
         }
 
-        $stripeCoupon = $this->get('stripe_client')
-            ->findCoupon($code);
+        try {
+            $stripeCoupon = $this->get('stripe_client')
+                ->findCoupon($code);
+        } catch (\Stripe\Error\InvalidRequest $e) {
+            $this->addFlash('error', 'Invalid coupon code!');
+
+            return $this->redirectToRoute('order_checkout');
+        }
 
         $this->get('shopping_cart')
             ->setCouponCode($code, $stripeCoupon->amount_off / 100);
