diff --git a/src/AppBundle/Controller/WebhookController.php b/src/AppBundle/Controller/WebhookController.php
index 4115480..8afe841 100644
--- a/src/AppBundle/Controller/WebhookController.php
+++ b/src/AppBundle/Controller/WebhookController.php
@@ -68,7 +68,13 @@ class WebhookController extends BaseController
 
                     if ($stripeEvent->data->object->attempt_count == 1) {
                         $user = $subscription->getUser();
+                        $stripeCustomer = $this->get('stripe_client')
+                            ->findCustomer($user->getStripeCustomerId());
+
+                        $hasCardOnFile = count($stripeCustomer->sources->data) > 0;
+
                         // todo - send the user an email about the problem
+                        // use hasCardOnFile to customize this
                     }
                 }
 
