diff --git a/app/Resources/views/order/checkout.html.twig b/app/Resources/views/order/checkout.html.twig
index 9d26e1b..a871998 100644
--- a/app/Resources/views/order/checkout.html.twig
+++ b/app/Resources/views/order/checkout.html.twig
@@ -5,6 +5,16 @@
     {{ parent() }}
 
     {{ include('order/_creditCardFormJavaScript.html.twig') }}
+
+    <script>
+        jQuery(document).ready(function() {
+            $('.js-show-code-form').on('click', function(e) {
+                e.preventDefault();
+
+                $('.js-code-form').show();
+            })
+        });
+    </script>
 {% endblock %}
 
 {% block body %}
@@ -44,6 +54,25 @@
                         </tr>
                     </tfoot>
                 </table>
+
+                <button class="btn btn-xs btn-link pull-right js-show-code-form">
+                    I have a coupon code
+                </button>
+
+                <div class="js-code-form" style="display: none;">
+                    <form action="" method="POST" class="form-inline">
+                        <div class="form-group">
+                            <div class="input-group">
+                                <span class="input-group-addon">
+                                    <i class="fa fa-terminal"></i>
+                                </span>
+                                <input type="text" name="code" autocomplete="off" class="form-control" placeholder="Coupon Code"/>
+                            </div>
+
+                            <button type="submit" class="btn btn-primary">Add</button>
+                        </div>
+                    </form>
+                </div>
             </div>
             <div class="col-xs-12 col-sm-6">
                 {{ include('order/_cardForm.html.twig') }}
