diff --git a/app/Resources/views/order/_cardForm.html.twig b/app/Resources/views/order/_cardForm.html.twig
index bdc1dc3..5c343da 100644
--- a/app/Resources/views/order/_cardForm.html.twig
+++ b/app/Resources/views/order/_cardForm.html.twig
@@ -53,7 +53,7 @@
 
     <div class="row">
         <div class="col-xs-8 col-sm-6 col-sm-offset-2 text-center">
-            <div class="alert alert-danger js-checkout-error hidden"></div>
+            <div class="alert alert-danger js-checkout-error {{ error ? '' : 'hidden' }}">{{ error }}</div>
         </div>
     </div>
 
diff --git a/src/AppBundle/Controller/OrderController.php b/src/AppBundle/Controller/OrderController.php
index 0de18e9..5995275 100644
--- a/src/AppBundle/Controller/OrderController.php
+++ b/src/AppBundle/Controller/OrderController.php
@@ -33,27 +33,36 @@ class OrderController extends BaseController
     {
         $products = $this->get('shopping_cart')->getProducts();
 
+        $error = false;
         if ($request->isMethod('POST')) {
             $token = $request->request->get('stripeToken');
 
-            $this->chargeCustomer($token);
+            try {
+                $this->chargeCustomer($token);
+            } catch (\Stripe\Error\Card $e) {
+                $error = 'There was a problem charging your card: '.$e->getMessage();
+            }
 
-            $this->get('shopping_cart')->emptyCart();
-            $this->addFlash('success', 'Order Complete! Yay!');
+            if (!$error) {
+                $this->get('shopping_cart')->emptyCart();
+                $this->addFlash('success', 'Order Complete! Yay!');
 
-            return $this->redirectToRoute('homepage');
+                return $this->redirectToRoute('homepage');
+            }
         }
 
         return $this->render('order/checkout.html.twig', array(
             'products' => $products,
             'cart' => $this->get('shopping_cart'),
-            'stripe_public_key' => $this->getParameter('stripe_public_key')
+            'stripe_public_key' => $this->getParameter('stripe_public_key'),
+            'error' => $error,
         ));
 
     }
 
     /**
      * @param $token
+     * @throws \Stripe\Error\Card
      */
     private function chargeCustomer($token)
     {
