diff --git a/src/AppBundle/Controller/OrderController.php b/src/AppBundle/Controller/OrderController.php
index 1aad14f..c5ca7ab 100644
--- a/src/AppBundle/Controller/OrderController.php
+++ b/src/AppBundle/Controller/OrderController.php
@@ -57,12 +57,14 @@ class OrderController extends BaseController
                 $customer->save();
             }
 
-            \Stripe\InvoiceItem::create(array(
-              "amount" => $this->get('shopping_cart')->getTotal() * 100,
-              "currency" => "usd",
-              "customer" => $user->getStripeCustomerId(),
-              "description" => "First test charge!"
-            ));
+            foreach ($this->get('shopping_cart')->getProducts() as $product) {
+                \Stripe\InvoiceItem::create(array(
+                    "amount" => $product->getPrice() * 100,
+                    "currency" => "usd",
+                    "customer" => $user->getStripeCustomerId(),
+                    "description" => $product->getName()
+                ));
+            }
             $invoice = \Stripe\Invoice::create(array(
                 "customer" => $user->getStripeCustomerId()
             ));
