diff --git a/app/config/services.yml b/app/config/services.yml
index f3b8a0c..f615a59 100644
--- a/app/config/services.yml
+++ b/app/config/services.yml
@@ -10,4 +10,5 @@ services:
 
     stripe_client:
         class: AppBundle\StripeClient
+        arguments: ['%stripe_secret_key%']
         autowire: true
diff --git a/src/AppBundle/Controller/OrderController.php b/src/AppBundle/Controller/OrderController.php
index 17d74f1..1117ec2 100644
--- a/src/AppBundle/Controller/OrderController.php
+++ b/src/AppBundle/Controller/OrderController.php
@@ -36,8 +36,6 @@ class OrderController extends BaseController
         if ($request->isMethod('POST')) {
             $token = $request->request->get('stripeToken');
 
-            \Stripe\Stripe::setApiKey($this->getParameter('stripe_secret_key'));
-
             $stripeClient = $this->get('stripe_client');
             /** @var User $user */
             $user = $this->getUser();
diff --git a/src/AppBundle/StripeClient.php b/src/AppBundle/StripeClient.php
index 23a5ecd..00683f6 100644
--- a/src/AppBundle/StripeClient.php
+++ b/src/AppBundle/StripeClient.php
@@ -9,9 +9,11 @@ class StripeClient
 {
     private $em;
 
-    public function __construct(EntityManager $em)
+    public function __construct($secretKey, EntityManager $em)
     {
         $this->em = $em;
+
+        \Stripe\Stripe::setApiKey($secretKey);
     }
 
     public function createCustomer(User $user, $paymentToken)
