diff --git a/src/AppBundle/Controller/OrderController.php b/src/AppBundle/Controller/OrderController.php
index fef9a51..17d74f1 100644
--- a/src/AppBundle/Controller/OrderController.php
+++ b/src/AppBundle/Controller/OrderController.php
@@ -44,10 +44,7 @@ class OrderController extends BaseController
             if (!$user->getStripeCustomerId()) {
                 $stripeClient->createCustomer($user, $token);
             } else {
-                $customer = \Stripe\Customer::retrieve($user->getStripeCustomerId());
-
-                $customer->source = $token;
-                $customer->save();
+                $stripeClient->updateCustomerCard($user, $token);
             }
 
             foreach ($this->get('shopping_cart')->getProducts() as $product) {
diff --git a/src/AppBundle/StripeClient.php b/src/AppBundle/StripeClient.php
index 1720669..23a5ecd 100644
--- a/src/AppBundle/StripeClient.php
+++ b/src/AppBundle/StripeClient.php
@@ -27,4 +27,12 @@ class StripeClient
 
         return $customer;
     }
+
+    public function updateCustomerCard(User $user, $paymentToken)
+    {
+        $customer = \Stripe\Customer::retrieve($user->getStripeCustomerId());
+
+        $customer->source = $paymentToken;
+        $customer->save();
+    }
 }
