diff --git a/app/config/parameters.yml.dist b/app/config/parameters.yml.dist
index 69e8170..c41b8af 100644
--- a/app/config/parameters.yml.dist
+++ b/app/config/parameters.yml.dist
@@ -19,3 +19,6 @@ parameters:
     secret:            ThisTokenIsNotSoSecretChangeIt
     stripe_secret_key: XXX_PRIVATEKEY_XXX
     stripe_public_key: YYY_PUBLISHABLE_KEY_YYY
+
+    # set to https on production
+    secure_scheme: http
diff --git a/src/AppBundle/Controller/OrderController.php b/src/AppBundle/Controller/OrderController.php
index 9b2b209..e43f446 100644
--- a/src/AppBundle/Controller/OrderController.php
+++ b/src/AppBundle/Controller/OrderController.php
@@ -27,7 +27,7 @@ class OrderController extends Controller
     }
 
     /**
-     * @Route("/checkout", name="order_checkout", schemes={"https"})
+     * @Route("/checkout", name="order_checkout", schemes={"%secure_scheme%"})
      * @Security("is_granted('ROLE_USER')")
      */
     public function checkoutAction(Request $request)
