diff --git a/app/Resources/views/product/pricing.html.twig b/app/Resources/views/product/pricing.html.twig
index 906e2c3..c146db7 100644
--- a/app/Resources/views/product/pricing.html.twig
+++ b/app/Resources/views/product/pricing.html.twig
@@ -16,7 +16,9 @@
                     <p class="price text-center">$99</p>
                     <p class="text-center">A fresh set of shears monthly along with our beloved after-shear in one of two fragrances!</p>
                 </div>
-                <a href="#" class="btn btn-warning center-block price-btn">
+                <a href="{{ path('order_add_subscription_to_cart', {
+                    'planId': 'TODO'
+                }) }}" class="btn btn-warning center-block price-btn">
                     Shear Me!
                 </a>
             </div>
@@ -29,7 +31,9 @@
                     <p class="price text-center">$199</p>
                     <p class="text-center">The perks of a Farmer Brent membership, but with a fresh bottle of conditioner and a comb each month!</p>
                 </div>
-                <a href="#" class="btn btn-warning center-block price-btn">
+                <a href="{{ path('order_add_subscription_to_cart', {
+                    'planId': 'TODO'
+                }) }}" class="btn btn-warning center-block price-btn">
                     Get Shearing!
                 </a>
             </div>
diff --git a/src/AppBundle/Controller/OrderController.php b/src/AppBundle/Controller/OrderController.php
index 5995275..c8dd3e8 100644
--- a/src/AppBundle/Controller/OrderController.php
+++ b/src/AppBundle/Controller/OrderController.php
@@ -26,6 +26,14 @@ class OrderController extends BaseController
     }
 
     /**
+     * @Route("/cart/subscription/{planId}", name="order_add_subscription_to_cart")
+     */
+    public function addSubscriptionToCartAction($planId)
+    {
+        // todo - add the subscription plan to the cart!
+    }
+
+    /**
      * @Route("/checkout", name="order_checkout", schemes={"%secure_channel%"})
      * @Security("is_granted('ROLE_USER')")
      */
diff --git a/src/AppBundle/Store/ShoppingCart.php b/src/AppBundle/Store/ShoppingCart.php
index ff17ee1..6063cce 100644
--- a/src/AppBundle/Store/ShoppingCart.php
+++ b/src/AppBundle/Store/ShoppingCart.php
@@ -3,22 +3,28 @@
 namespace AppBundle\Store;
 
 use AppBundle\Entity\Product;
+use AppBundle\Subscription\SubscriptionHelper;
 use Doctrine\ORM\EntityManager;
 use Symfony\Component\HttpFoundation\Session\Session;
 
 class ShoppingCart
 {
     const CART_PRODUCTS_KEY = '_shopping_cart.products';
+    const CART_PLAN_KEY = '_shopping_cart.subscription_plan';
+    const CART_COUPON_CODE_KEY = '_shopping_cart.coupon_code';
+    const CART_COUPON_VALUE_KEY = '_shopping_cart.coupon_value';
 
     private $session;
     private $em;
+    private $subscriptionHelper;
 
     private $products;
 
-    public function __construct(Session $session, EntityManager $em)
+    public function __construct(Session $session, EntityManager $em, SubscriptionHelper $subscriptionHelper)
     {
         $this->session = $session;
         $this->em = $em;
+        $this->subscriptionHelper = $subscriptionHelper;
     }
 
     public function addProduct(Product $product)
@@ -32,6 +38,14 @@ class ShoppingCart
         $this->updateProducts($products);
     }
 
+    public function addSubscription($planId)
+    {
+        $this->session->set(
+            self::CART_PLAN_KEY,
+            $planId
+        );
+    }
+
     /**
      * @return Product[]
      */
@@ -56,6 +70,17 @@ class ShoppingCart
         return $this->products;
     }
 
+    /**
+     * @return \AppBundle\Subscription\SubscriptionPlan|null
+     */
+    public function getSubscriptionPlan()
+    {
+        $planId = $this->session->get(self::CART_PLAN_KEY);
+
+        return $this->subscriptionHelper
+            ->findPlan($planId);
+    }
+
     public function getTotal()
     {
         $total = 0;
@@ -63,12 +88,49 @@ class ShoppingCart
             $total += $product->getPrice();
         }
 
+        if ($this->getSubscriptionPlan()) {
+            $price = $this->getSubscriptionPlan()
+                ->getPrice();
+
+            $total += $price;
+        }
+
         return $total;
     }
 
+    public function getTotalWithDiscount()
+    {
+        return max($this->getTotal() - $this->getCouponCodeValue(), 0);
+    }
+
+    public function setCouponCode($code, $value)
+    {
+        $this->session->set(
+            self::CART_COUPON_CODE_KEY,
+            $code
+        );
+
+        $this->session->set(
+            self::CART_COUPON_VALUE_KEY,
+            $value
+        );
+    }
+
+    public function getCouponCode()
+    {
+        return $this->session->get(self::CART_COUPON_CODE_KEY);
+    }
+
+    public function getCouponCodeValue()
+    {
+        return $this->session->get(self::CART_COUPON_VALUE_KEY);
+    }
+
     public function emptyCart()
     {
         $this->updateProducts([]);
+        $this->updatePlanId(null);
+        $this->setCouponCode(null, null);
     }
 
     /**
@@ -84,4 +146,9 @@ class ShoppingCart
 
         $this->session->set(self::CART_PRODUCTS_KEY, $ids);
     }
-}
\ No newline at end of file
+
+    private function updatePlanId($planId)
+    {
+        $this->session->set(self::CART_PLAN_KEY, $planId);
+    }
+}
