diff --git a/app/Resources/views/profile/account.html.twig b/app/Resources/views/profile/account.html.twig
index c40d220..854e5e9 100644
--- a/app/Resources/views/profile/account.html.twig
+++ b/app/Resources/views/profile/account.html.twig
@@ -16,7 +16,11 @@
                         <tr>
                             <th>Subscription</th>
                             <td>
-                                None
+                                {% if app.user.subscription %}
+                                    <span class="label label-success">Active</span>
+                                {% else %}
+                                    <span class="label label-default">None</span>
+                                {% endif %}
                             </td>
                         </tr>
                         <tr>
