diff --git a/src/AppBundle/Controller/OrderController.php b/src/AppBundle/Controller/OrderController.php
index f15c4e9..d36dcec 100644
--- a/src/AppBundle/Controller/OrderController.php
+++ b/src/AppBundle/Controller/OrderController.php
@@ -104,10 +104,15 @@ class OrderController extends BaseController
 
         if ($cart->getSubscriptionPlan()) {
             // a subscription creates an invoice
-            $stripeClient->createSubscription(
+            $stripeSubscription = $stripeClient->createSubscription(
                 $user,
                 $cart->getSubscriptionPlan()
             );
+
+            $this->get('subscription_helper')->addSubscriptionToUser(
+                $stripeSubscription,
+                $user
+            );
         } else {
             // charge the invoice!
             $stripeClient->createInvoice($user, true);
diff --git a/src/AppBundle/Entity/Subscription.php b/src/AppBundle/Entity/Subscription.php
index 438f62c..d81cd06 100644
--- a/src/AppBundle/Entity/Subscription.php
+++ b/src/AppBundle/Entity/Subscription.php
@@ -91,4 +91,11 @@ class Subscription
     {
         return $this->billingPeriodEndsAt;
     }
+
+    public function activateSubscription($stripePlanId, $stripeSubscriptionId)
+    {
+        $this->stripePlanId = $stripePlanId;
+        $this->stripeSubscriptionId = $stripeSubscriptionId;
+        $this->endsAt = null;
+    }
 }
diff --git a/src/AppBundle/Subscription/SubscriptionHelper.php b/src/AppBundle/Subscription/SubscriptionHelper.php
index c2685e9..48e394b 100644
--- a/src/AppBundle/Subscription/SubscriptionHelper.php
+++ b/src/AppBundle/Subscription/SubscriptionHelper.php
@@ -2,13 +2,21 @@
 
 namespace AppBundle\Subscription;
 
+use AppBundle\Entity\Subscription;
+use AppBundle\Entity\User;
+use Doctrine\ORM\EntityManager;
+
 class SubscriptionHelper
 {
     /** @var SubscriptionPlan[] */
     private $plans = [];
 
-    public function __construct()
+    private $em;
+
+    public function __construct(EntityManager $em)
     {
+        $this->em = $em;
+
         $this->plans[] = new SubscriptionPlan(
             'farmer_brent_monthly',
             'Farmer Brent',
@@ -34,4 +42,21 @@ class SubscriptionHelper
             }
         }
     }
+
+    public function addSubscriptionToUser(\Stripe\Subscription $stripeSubscription, User $user)
+    {
+        $subscription = $user->getSubscription();
+        if (!$subscription) {
+            $subscription = new Subscription();
+            $subscription->setUser($user);
+        }
+
+        $subscription->activateSubscription(
+            $stripeSubscription->plan->id,
+            $stripeSubscription->id
+        );
+
+        $this->em->persist($subscription);
+        $this->em->flush($subscription);
+    }
 }
