diff --git a/app/Resources/views/base.html.twig b/app/Resources/views/base.html.twig
index 6c5df07..f2254fc 100644
--- a/app/Resources/views/base.html.twig
+++ b/app/Resources/views/base.html.twig
@@ -33,6 +33,9 @@
                     <ul class="nav navbar-nav">
                         {% if is_granted('ROLE_USER') %}
                             <li>
+                                <a href="{{ path('profile_account') }}">Account</a>
+                            </li>
+                            <li>
                                 <a href="{{ path('fos_user_security_logout') }}">Logout</a>
                             </li>
                         {% else %}
diff --git a/app/Resources/views/profile/account.html.twig b/app/Resources/views/profile/account.html.twig
new file mode 100644
index 0000000..c40d220
--- /dev/null
+++ b/app/Resources/views/profile/account.html.twig
@@ -0,0 +1,44 @@
+{% extends 'base.html.twig' %}
+
+{% block body %}
+<div class="nav-space">
+    <div class="container">
+        <div class="row">
+            <div class="col-xs-12">
+                <h1>My Account</h1>
+            </div>
+        </div>
+
+        <div class="row">
+            <div class="col-xs-6">
+                <table class="table">
+                    <tbody>
+                        <tr>
+                            <th>Subscription</th>
+                            <td>
+                                None
+                            </td>
+                        </tr>
+                        <tr>
+                            <th>Next Billing at:</th>
+                            <td>
+                                n/a
+                            </td>
+                        </tr>
+                        <tr>
+                            <th>Credit Card</th>
+                            <td>
+                                None
+                            </td>
+                        </tr>
+                    </tbody>
+                </table>
+            </div>
+            <div class="col-xs-6">
+
+            </div>
+        </div>
+    </div>
+</div>
+{% endblock %}
+
diff --git a/src/AppBundle/Controller/ProfileController.php b/src/AppBundle/Controller/ProfileController.php
new file mode 100644
index 0000000..c9d2c11
--- /dev/null
+++ b/src/AppBundle/Controller/ProfileController.php
@@ -0,0 +1,20 @@
+<?php
+
+namespace AppBundle\Controller;
+
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Security;
+
+/**
+ * @Security("is_granted('ROLE_USER')")
+ */
+class ProfileController extends BaseController
+{
+    /**
+     * @Route("/profile", name="profile_account")
+     */
+    public function accountAction()
+    {
+        return $this->render('profile/account.html.twig');
+    }
+}
