diff --git a/app/Resources/views/order/_creditCardFormJavaScript.html.twig b/app/Resources/views/order/_creditCardFormJavaScript.html.twig
new file mode 100644
index 0000000..b934282
--- /dev/null
+++ b/app/Resources/views/order/_creditCardFormJavaScript.html.twig
@@ -0,0 +1,51 @@
+<script type="text/javascript" src="https://js.stripe.com/v2/"></script>
+<script src="{{ asset('js/jquery.payment.min.js') }}"></script>
+
+<script type="text/javascript">
+    Stripe.setPublishableKey('{{ stripe_public_key }}');
+
+    $(function () {
+        var $form = $('.js-checkout-form');
+
+        $form.find('.js-cc-number').payment('formatCardNumber');
+        $form.find('.js-cc-exp').payment('formatCardExpiry');
+        $form.find('.js-cc-cvc').payment('formatCardCVC');
+
+        $form.submit(function (event) {
+            event.preventDefault();
+
+            // Disable the submit button to prevent repeated clicks:
+            $form.find('.js-submit-button').prop('disabled', true);
+
+            // Request a token from Stripe:
+            Stripe.card.createToken($form, stripeResponseHandler);
+        });
+    });
+
+    function stripeResponseHandler(status, response) {
+        // Grab the form:
+        var $form = $('.js-checkout-form');
+
+        if (response.error) { // Problem!
+
+            // Show the errors on the form:
+            $form.find('.js-checkout-error')
+                .text(response.error.message)
+                .removeClass('hidden');
+            $form.find('.js-submit-button').prop('disabled', false); // Re-enable submission
+
+        } else { // Token was created!
+            $form.find('.js-checkout-error')
+                .addClass('hidden');
+
+            // Get the token ID:
+            var token = response.id;
+
+            // Insert the token ID into the form so it gets submitted to the server:
+            $form.append($('<input type="hidden" name="stripeToken">').val(token));
+
+            // Submit the form:
+            $form.get(0).submit();
+        }
+    }
+</script>
\ No newline at end of file
diff --git a/app/Resources/views/order/checkout.html.twig b/app/Resources/views/order/checkout.html.twig
index ddc8a17..9d26e1b 100644
--- a/app/Resources/views/order/checkout.html.twig
+++ b/app/Resources/views/order/checkout.html.twig
@@ -4,57 +4,7 @@
 {% block javascripts %}
     {{ parent() }}
 
-    <script type="text/javascript" src="https://js.stripe.com/v2/"></script>
-    <script src="{{ asset('js/jquery.payment.min.js') }}"></script>
-
-    <script type="text/javascript">
-        Stripe.setPublishableKey('{{ stripe_public_key }}');
-
-        $(function () {
-            var $form = $('.js-checkout-form');
-
-            $form.find('.js-cc-number').payment('formatCardNumber');
-            $form.find('.js-cc-exp').payment('formatCardExpiry');
-            $form.find('.js-cc-cvc').payment('formatCardCVC');
-
-            $form.submit(function (event) {
-                event.preventDefault();
-
-                // Disable the submit button to prevent repeated clicks:
-                $form.find('.js-submit-button').prop('disabled', true);
-
-                // Request a token from Stripe:
-                Stripe.card.createToken($form, stripeResponseHandler);
-            });
-        });
-
-        function stripeResponseHandler(status, response) {
-            // Grab the form:
-            var $form = $('.js-checkout-form');
-
-            if (response.error) { // Problem!
-
-                // Show the errors on the form:
-                $form.find('.js-checkout-error')
-                    .text(response.error.message)
-                    .removeClass('hidden');
-                $form.find('.js-submit-button').prop('disabled', false); // Re-enable submission
-
-            } else { // Token was created!
-                $form.find('.js-checkout-error')
-                    .addClass('hidden');
-
-                // Get the token ID:
-                var token = response.id;
-
-                // Insert the token ID into the form so it gets submitted to the server:
-                $form.append($('<input type="hidden" name="stripeToken">').val(token));
-
-                // Submit the form:
-                $form.get(0).submit();
-            }
-        }
-    </script>
+    {{ include('order/_creditCardFormJavaScript.html.twig') }}
 {% endblock %}
 
 {% block body %}
diff --git a/app/Resources/views/profile/account.html.twig b/app/Resources/views/profile/account.html.twig
index 3790832..f576b12 100644
--- a/app/Resources/views/profile/account.html.twig
+++ b/app/Resources/views/profile/account.html.twig
@@ -3,6 +3,8 @@
 {% block javascripts %}
     {{ parent() }}
 
+    {{ include('order/_creditCardFormJavaScript.html.twig') }}
+
     <script>
         jQuery(document).ready(function() {
             $('.js-open-credit-card-form').on('click', function(e) {
diff --git a/src/AppBundle/Controller/ProfileController.php b/src/AppBundle/Controller/ProfileController.php
index f798cdf..ca6135a 100644
--- a/src/AppBundle/Controller/ProfileController.php
+++ b/src/AppBundle/Controller/ProfileController.php
@@ -17,7 +17,8 @@ class ProfileController extends BaseController
     public function accountAction()
     {
         return $this->render('profile/account.html.twig', [
-            'error' => null
+            'error' => null,
+            'stripe_public_key' => $this->getParameter('stripe_public_key'),
         ]);
     }
 
