diff --git a/app/Resources/views/profile/account.html.twig b/app/Resources/views/profile/account.html.twig
index 2804dac..c681756 100644
--- a/app/Resources/views/profile/account.html.twig
+++ b/app/Resources/views/profile/account.html.twig
@@ -12,6 +12,12 @@
 
                 $('.js-update-card-wrapper').slideToggle();
             });
+
+            $('.js-change-plan-button').on('click', function(e) {
+                e.preventDefault();
+
+                swal('Loading Plan Details...');
+            })
         });
     </script>
 {% endblock %}
@@ -51,6 +57,10 @@
                                         {{ currentPlan.name }}
 
                                         <span class="label label-success">Active</span>
+
+                                        <button class="btn btn-xs btn-link pull-right js-change-plan-button" data-plan-id="{{ otherPlan.planId }}" data-plan-name="{{ otherPlan.name }}">
+                                            Change to {{ otherPlan.name }}
+                                        </button>
                                     {% endif %}
                                 {% else %}
                                     <span class="label label-default">None</span>
diff --git a/src/AppBundle/Controller/ProfileController.php b/src/AppBundle/Controller/ProfileController.php
index 5d6ec40..87db981 100644
--- a/src/AppBundle/Controller/ProfileController.php
+++ b/src/AppBundle/Controller/ProfileController.php
@@ -18,15 +18,20 @@ class ProfileController extends BaseController
     public function accountAction()
     {
         $currentPlan = null;
+        $otherPlan = null;
         if ($this->getUser()->hasActiveSubscription()) {
             $currentPlan = $this->get('subscription_helper')
                 ->findPlan($this->getUser()->getSubscription()->getStripePlanId());
+
+            $otherPlan = $this->get('subscription_helper')
+                ->findPlanToChangeTo($currentPlan->getPlanId());
         }
 
         return $this->render('profile/account.html.twig', [
             'error' => null,
             'stripe_public_key' => $this->getParameter('stripe_public_key'),
-            'currentPlan' => $currentPlan
+            'currentPlan' => $currentPlan,
+            'otherPlan' => $otherPlan,
         ]);
     }
 
diff --git a/src/AppBundle/Subscription/SubscriptionHelper.php b/src/AppBundle/Subscription/SubscriptionHelper.php
index 3f9fe07..35524dc 100644
--- a/src/AppBundle/Subscription/SubscriptionHelper.php
+++ b/src/AppBundle/Subscription/SubscriptionHelper.php
@@ -43,6 +43,21 @@ class SubscriptionHelper
         }
     }
 
+    /**
+     * @param $currentPlanId
+     * @return SubscriptionPlan
+     */
+    public function findPlanToChangeTo($currentPlanId)
+    {
+        if (strpos($currentPlanId, 'farmer_brent') !== false) {
+            $newPlanId = str_replace('farmer_brent', 'new_zealander', $currentPlanId);
+        } else {
+            $newPlanId = str_replace('new_zealander', 'farmer_brent', $currentPlanId);
+        }
+
+        return $this->findPlan($newPlanId);
+    }
+
     public function addSubscriptionToUser(\Stripe\Subscription $stripeSubscription, User $user)
     {
         $subscription = $user->getSubscription();
