diff --git a/src/AppBundle/Controller/WebhookController.php b/src/AppBundle/Controller/WebhookController.php
index 3abade3..1dd1534 100644
--- a/src/AppBundle/Controller/WebhookController.php
+++ b/src/AppBundle/Controller/WebhookController.php
@@ -48,6 +48,19 @@ class WebhookController extends BaseController
                     $subscriptionHelper->handleSubscriptionPaid($subscription, $stripeSubscription);
                 }
                 break;
+            case 'invoice.payment_failed':
+                $stripeSubscriptionId = $stripeEvent->data->object->subscription;
+
+                if ($stripeSubscriptionId) {
+                    $subscription = $this->findSubscription($stripeSubscriptionId);
+
+                    if ($stripeEvent->data->object->attempt_count == 1) {
+                        $user = $subscription->getUser();
+                        // todo - send the user an email about the problem
+                    }
+                }
+
+                break;
             default:
                 // allow this - we'll have Stripe send us everything
                 // throw new \Exception('Unexpected webhook type form Stripe! '.$stripeEvent->type);
