diff --git a/src/AppBundle/Controller/WebhookController.php b/src/AppBundle/Controller/WebhookController.php
index d1adb38..3abade3 100644
--- a/src/AppBundle/Controller/WebhookController.php
+++ b/src/AppBundle/Controller/WebhookController.php
@@ -37,8 +37,20 @@ class WebhookController extends BaseController
                 $subscriptionHelper->fullyCancelSubscription($subscription);
 
                 break;
+            case 'invoice.payment_succeeded':
+                $stripeSubscriptionId = $stripeEvent->data->object->subscription;
+
+                if ($stripeSubscriptionId) {
+                    $subscription = $this->findSubscription($stripeSubscriptionId);
+                    $stripeSubscription = $this->get('stripe_client')
+                        ->findSubscription($stripeSubscriptionId);
+
+                    $subscriptionHelper->handleSubscriptionPaid($subscription, $stripeSubscription);
+                }
+                break;
             default:
-                throw new \Exception('Unexpected webhook type form Stripe! '.$stripeEvent->type);
+                // allow this - we'll have Stripe send us everything
+                // throw new \Exception('Unexpected webhook type form Stripe! '.$stripeEvent->type);
         }
 
         return new Response('Event Handled: '.$stripeEvent->type);
@@ -63,4 +75,4 @@ class WebhookController extends BaseController
 
         return $subscription;
     }
-}
\ No newline at end of file
+}
diff --git a/src/AppBundle/Entity/Subscription.php b/src/AppBundle/Entity/Subscription.php
index 2c6a897..e3b75be 100644
--- a/src/AppBundle/Entity/Subscription.php
+++ b/src/AppBundle/Entity/Subscription.php
@@ -127,4 +127,9 @@ class Subscription
     {
         return $this->endsAt !== null;
     }
+
+    public function setBillingPeriodEndsAt($billingPeriodEndsAt)
+    {
+        $this->billingPeriodEndsAt = $billingPeriodEndsAt;
+    }
 }
diff --git a/src/AppBundle/StripeClient.php b/src/AppBundle/StripeClient.php
index e976c6c..2ef26fe 100644
--- a/src/AppBundle/StripeClient.php
+++ b/src/AppBundle/StripeClient.php
@@ -123,4 +123,13 @@ class StripeClient
     {
         return \Stripe\Event::retrieve($eventId);
     }
+
+    /**
+     * @param $stripeSubscriptionId
+     * @return \Stripe\Subscription
+     */
+    public function findSubscription($stripeSubscriptionId)
+    {
+        return \Stripe\Subscription::retrieve($stripeSubscriptionId);
+    }
 }
diff --git a/src/AppBundle/Subscription/SubscriptionHelper.php b/src/AppBundle/Subscription/SubscriptionHelper.php
index 1b921a2..141c6d0 100644
--- a/src/AppBundle/Subscription/SubscriptionHelper.php
+++ b/src/AppBundle/Subscription/SubscriptionHelper.php
@@ -77,4 +77,13 @@ class SubscriptionHelper
         $this->em->persist($subscription);
         $this->em->flush($subscription);
     }
+
+    public function handleSubscriptionPaid(Subscription $subscription, \Stripe\Subscription $stripeSubscription)
+    {
+        $newPeriodEnd = \DateTime::createFromFormat('U', $stripeSubscription->current_period_end);
+
+        $subscription->setBillingPeriodEndsAt($newPeriodEnd);
+        $this->em->persist($subscription);
+        $this->em->flush($subscription);
+    }
 }
