diff --git a/src/AppBundle/Subscription/SubscriptionHelper.php b/src/AppBundle/Subscription/SubscriptionHelper.php
index 141c6d0..3f9fe07 100644
--- a/src/AppBundle/Subscription/SubscriptionHelper.php
+++ b/src/AppBundle/Subscription/SubscriptionHelper.php
@@ -82,6 +82,9 @@ class SubscriptionHelper
     {
         $newPeriodEnd = \DateTime::createFromFormat('U', $stripeSubscription->current_period_end);
 
+        // you can use this to send emails to new or renewal customers
+        $isRenewal = $newPeriodEnd > $subscription->getBillingPeriodEndsAt();
+
         $subscription->setBillingPeriodEndsAt($newPeriodEnd);
         $this->em->persist($subscription);
         $this->em->flush($subscription);
